components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/utils/mappings/APIMappingUtil.java

```diff
@@ -668,19 +672,67 @@ private static void setEndpointSecurityFromApiDTOToModel (APIDetailedDTO dto, AP
             securityDTO.setType(APIEndpointSecurityDTO.TypeEnum.basic); //set default as basic
             securityDTO.setUsername(api.getEndpointUTUsername());
-            securityDTO.setPassword(""); // Do not expose password
+            String tenantDomain = MultitenantUtils.getTenantDomain(APIUtil.replaceEmailDomainBack(api.getId()
+                    .getProviderName()));
+            if (checkEndpointSecurityPasswordEnabled(tenantDomain)) {
+                securityDTO.setPassword(api.getEndpointUTPassword());
+            } else {
+                securityDTO.setPassword(""); //Do not expose password
+            }
             if (api.isEndpointAuthDigest()) {
                 securityDTO.setType(APIEndpointSecurityDTO.TypeEnum.digest);
             }
             dto.setEndpointSecurity(securityDTO);
         }
     }
 
+    /**
+     * This method used to check whether the config for exposing endpoint security password when getting API is enabled
+     * or not in tenant-config.json in registry.
+     *
+     * @return boolean as config enabled or not
+     * @throws APIManagementException
+     */
+    private static boolean checkEndpointSecurityPasswordEnabled(String tenantDomainName) throws APIManagementException {
+        JSONObject apiTenantConfig;
+        try {
+            APIMRegistryServiceImpl apimRegistryService = new APIMRegistryServiceImpl();
+            String content = apimRegistryService.getConfigRegistryResourceContent(tenantDomainName,
+                    APIConstants.API_TENANT_CONF_LOCATION);
+            if (content != null) {
+                JSONParser parser = new JSONParser();
+                apiTenantConfig = (JSONObject) parser.parse(content);
+                if (apiTenantConfig != null) {
+                    Object value = apiTenantConfig.get(APIConstants.API_TENANT_CONF_EXPOSE_ENDPOINT_PASSWORD);
+                    if (value != null) {
+                        return Boolean.parseBoolean(value.toString());
+                    }
+                }
+            }
+        } catch (UserStoreException e) {
+            String msg = "UserStoreException thrown when getting API tenant config from registry while reading " +
+                    "ExposeEndpointPassword config";
+            log.error(msg, e);
+            throw new APIManagementException(msg, e);
+        } catch (RegistryException e) {
+            String msg = "RegistryException thrown when getting API tenant config from registry while reading " +
+                    "ExposeEndpointPassword config";
+            log.error(msg, e);
+            throw new APIManagementException(msg, e);
+        } catch (ParseException e) {
+            String msg = "ParseException thrown when parsing API tenant config from registry while reading " +
+                    "ExposeEndpointPassword config";
+            log.error(msg, e);
+            throw new APIManagementException(msg, e);
```

Comment:
We are doing both logging and throwing Exception. Won't that lead to duplicate error logs?

Reply:
yes the error is getting logged again at upper level. Thank you for pointing it out. Fixed it with https://github.com/wso2/carbon-apimgt/pull/5919/files